app/commands/git/pull_request/create_or_update.rb

```diff
@@ -0,0 +1,58 @@
+module Git
+  class PullRequest
+    class CreateOrUpdate
+      include Mandate
+
+      initialize_with :node_id, :attributes
+
+      def call
+        pull_request = ::Git::PullRequest.create_or_find_by!(node_id: node_id) do |pr|
+          pr.number = attributes[:pr_number]
+          pr.repo = attributes[:repo]
+          pr.author_github_username = attributes[:author]
+          pr.merged_by_github_username = attributes[:merged_by]
+          pr.data = attributes[:data]
+        end
+
+        pull_request.update!(
+          number: attributes[:pr_number],
+          repo: attributes[:repo],
+          author_github_username: attributes[:author],
+          merged_by_github_username: attributes[:merged_by],
+          data: attributes[:data],
+          reviews: reviews(pull_request)
+        )
+
+        pull_request
+      end
+
+      private
+      def reviews(pull_request)
+        attributes[:reviews].to_a.map do |review|
+          Git::PullRequestReview::Create.(
+            pull_request,
+            review[:node_id],
+            reviewer_github_username: review[:reviewer]
+          )
+        end
+      end
+
+      def data
+        {
+          action: attributes[:action],
+          author: attributes[:login],
+          url: attributes[:url],
+          html_url: attributes[:html_url],
+          labels: attributes[:labels],
+          repo: attributes[:repo],
+          pr_id: attributes[:pr_id],
+          pr_number: attributes[:pr_number],
+          merged: attributes[:merged],
+          merged_by: attributes[:merged_by],
+          state: attributes[:state],
+          reviews: attributes[:reviews]
+        }
```

Comment:
For keys like `merged_by` and `author`, should we put a type? I don't know if those are ids or handles etc at a glance. So `merged_by_id` or `merged_by_handle` etc.

Reply:
(This is a general question/comment for the whole PR I guess)

Reply:
I've just seen that you do that above: https://github.com/exercism/website/pull/567/files#diff-4ac666ae1e265a5121c154d15352029fa23064598dae8316ae2561ab31f0fcc3R13

So maybe that comment is redundant.

But then what made you choose to change `attributes[:login]` to be called `author`?

Reply:
This is an inconsistency. I'll update the data properties to be proper names.